### PR TITLE
Feature/google cloud provider

### PR DESCRIPTION
**Pull request by mturetskii**
### Description of the Change

This pull request introduces support for Google Cloud Platform (GCP), enabling our platform to leverage a wide range of cloud services provided by Google.

### CODE REVIEW

File: tools/cli/commands/setup.py [hunk 1]
```diff
@@ -1052,3 +1085,14 @@ def setup_param_validator(params: StateStore) -> bool:
             return False
 
     return True
+
+
+def get_dns_deployment_name_by_cloud_provider(cloud_provider) -> str:
```
Comment (all4code) [inline tools/cli/commands/setup.py:1090]: It's a good candidate to go under cloud provider manager "interface"

**Review by all4code — DISMISSED** [review]

**Review by sergs-pci — APPROVED** [review]

File: platform/terraform/modules/cloud_gcp/kms.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
+resource "google_kms_key_ring" "vault_key_ring" {
+  name     = "${local.name}-vault-${random_string.uniq_kms_suffix.result}"
+  location = "global"
+  project  = local.project_id
+}
+
+resource "google_kms_crypto_key" "vault_unseal_key" {
+  name            = "vault-unseal"
+  key_ring        = google_kms_key_ring.vault_key_ring.id
+  rotation_period = "100000s"
```
[the remark this answers is not in the document]
Comment (sergs-pci) [reply platform/terraform/modules/cloud_gcp/kms.tf:10]: We can't use 24h here,  It must be greater than a day (ie, 86400).

[the remark this answers is not in the document]
Comment (mturetskii) [reply platform/terraform/modules/cloud_gcp/kms.tf]: Done. Made it 365d.

File: tools/cli/services/cloud/gcp/gcp_manager.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    def account(self) -> str:
+        """AWS account id"""
+        return self._gcp_sdk.project_id
+
+    @trace()
+    def protect_iac_state_storage(self, name: str, identity: str):
+        """
+        Restrict access to cloud native terraform remote state storage
+        """
+        pass
```
[the remark this answers is not in the document]
Comment (mturetskii) [reply tools/cli/services/cloud/gcp/gcp_manager.py]: Done. Found a way, it should work fine in our case.

File: tools/cli/commands/setup.py (hunk 1 above)
Comment (mturetskii) [reply tools/cli/commands/setup.py]: Done, moved to cloud manager class.

**Review by all4code — APPROVED** [review]

**Pull request merged**